src/remoted/secure.c

```diff
@@ -455,14 +458,38 @@ int _close_sock(keystore * keys, int sock) {
 
 int key_request_connect() {
 #ifndef WIN32
-    return OS_ConnectUnixDomain(isChroot() ? WM_KEY_REQUEST_SOCK : WM_KEY_REQUEST_SOCK_PATH, SOCK_DGRAM, OS_MAXSTR);
+    if(logr.mode == KEYPOLL_MODE_MASTER)
+        return OS_ConnectUnixDomain(WM_KEY_REQUEST_CLUSTER_SOCK, SOCK_STREAM, OS_MAXSTR);
+    else
+        return OS_ConnectUnixDomain(WM_KEY_REQUEST_SOCK, SOCK_DGRAM, OS_MAXSTR);
 #else
     return -1;
 #endif
 }
 
 static int send_key_request(int socket,const char *msg) {
-    return OS_SendUnix(socket,msg,strlen(msg));
+
+    if(logr.mode == KEYPOLL_MODE_MASTER){
+        const char * message = "{\"message\":\"";
+        int size_payload = strlen(message)+strlen(msg)+3;
+        char * payload = malloc(size_payload);
+        snprintf(payload, size_payload, "{\"message\":\"%s\"}", msg);
+        int rc = OS_SendSecureTCPCluster(socket, "run_keypoll", payload, strlen(payload));
+        char buffer[OS_MAXSTR + 1];
+        int recv_msg = OS_RecvSecureClusterTCP(socket, buffer,OS_MAXSTR + 1);
```

Comment:
We must call `OS_RecvSecureClusterTCP()` if and only if `OS_SendSecureTCPCluster()` was successful.